File: session_2/ruby/src/doubles.rb
class Dummy
  def register_client(client)
  end
end

class Stub
  def register_client(client)
  end

  def do_something_important
    'OK'
  end
end

class Mock
  def initialize
    @calls = []
  end

  def register_client(client)
  end

  def do_something_important
    @calls << 'do_something_important'
  end

  def verify(call)
    @calls.include?(call)
  end
end

File: session_2/php/tests/Doubles.php
<?php

class Dummy
{
  public function registerClient($client)
  {

  }
}

class Stub
{
  public function registerClient($client)
  {

  }

  public function doSomethingImportant()
  {
    return 'OK';
  }
}

class Mock
{
  private $calls;

  public function __construct()
  {
    $this->calls = [];
  }

  public function registerClient($client)
  {

  }

  public function doSomethingImportant()
  {
    array_push($this->calls, 'doSomethingImportant');
  }

  public function verify($call)
  {
    return in_array($call, $this->calls);
  }
}
